src/renderer/components/status-bar/status-bar-items.injectable.tsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import type { IComputedValue } from "mobx";
+import { computed } from "mobx";
+import type { StatusBarItemProps } from "./status-bar-registration";
+import type { StatusBarItem } from "./status-bar-item-injection-token";
+import { statusBarItemInjectionToken } from "./status-bar-item-injection-token";
+import { computedInjectManyInjectable } from "@ogre-tools/injectable-extension-for-mobx";
+
+export interface StatusBarItems {
+  right: React.ComponentType<StatusBarItemProps>[];
+  left: React.ComponentType<StatusBarItemProps>[];
+}
+
+interface Dependencies {
+  registrations: IComputedValue<StatusBarItem[]>;
+}
+
+function getStatusBarItems({ registrations }: Dependencies): IComputedValue<StatusBarItems> {
+  return computed(() => {
+    const res: StatusBarItems = {
+      left: [],
+      right: [],
+    };
+
+    for (const registration of registrations.get()) {
+      if (!registration || typeof registration !== "object") {
+        continue;
+      }
```

Comment:
What would be case that registration does not exist or wouldn't be object? Stuff coming from extension api? If so, this belongs to `registrator`. In our code base we shouldn't need it, since the interface is enforced.

Reply:
Also wouldn't it make sense to throw in this case? So that extension developer gets the feedback immediately.